### ISSUE TITLE
Feature Request: Being able to only reload the graph

### ISSUE THREAD

**Issue by mmende** [alphacep/vosk-api#473]
First of all, thanks for publishing this nice API. I really appreciate the well-thought-out API-design and hope I can contribute to it here and there.

To the topic: Let's say I have built certain domain specific graphs or graphs with an extended vocabulary or virtually anything language model specific. In these cases it would be nice to be able to reuse the already loaded acoustic model but only load the graph specific stuff as adjusted Model instance. This could for example be realized by adding something like `Model loadGraph(const char *graph_path);` or `Model loadGraph(const char *graph_path, const char *rescore_path);` to the Model class. Especially in a server environment where potentially lots of models / graphs are preloaded, this feature would be very useful in terms of memory footprint.

How do you think about that?

**Comment by nshmyrev** [comment]
Model only holds the static data. You can update the grammar in recognizer constructor:

https://github.com/alphacep/vosk-api/blob/db41f16051bd1f5ed7d44b9a69280321c875da7c/python/example/test_words.py#L20

we can add a method to load graph from the file too.

**Comment by mmende** [comment]
Yes, I already saw the possibility to update the grammar this way which seems very helpful. However, at the moment this doesn't allow adding new OOV words which would be possible by using a customized graph for example. Another advantage of using customized graphs over the default one would be, that the weights can be tuned for the specific domain.

I can make a pull request if you like.
Kind regards

**Comment by nshmyrev** [comment]
> Another advantage of using customized graphs over the default one would be, that the weights can be tuned for the specific domain.

It would be great to have another recognizer constructor that reads graph from a file or from a wrap of fst.
